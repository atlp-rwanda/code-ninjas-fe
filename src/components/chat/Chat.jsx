import React, { useState, useEffect } from 'react';
import { makeStyles } from '@mui/styles';
import './Chat.scss';
import jwtDecode from 'jwt-decode';
import io from 'socket.io-client';
import { Formik, Form } from 'formik';
import { TextField, Button, Grid } from '@material-ui/core';
import Conversation from './conversations/Conversations';
import Message from './message/Message';
import ChatOnline from './chatOnline/ChatOnline';

const useStyles = makeStyles(() => ({
  container: {
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
}));

const socket = io.connect('http://localhost:3000');
// const token = localStorage.getItem('token');
// const { user } = jwtDecode(token);
// const username = user.userName;
// socket.emit('joinRoom', { username });

function Chat() {
  const [currentMessage, setCurrentMessage] = useState('');
  const [messageData, setMessageData] = useState('');
  const [messageList, setMessageList] = useState([]);

  const sendMessage = (e) => {
    e.preventDefault();

    // Get message text
    let msg = currentMessage;

    msg = msg.trim();

    if (!msg) {
      return false;
    }

    // Emit message to server
    socket.emit('chatMessage', msg);

    // Clear input
    e.target.elements.msg.value = '';
    e.target.elements.msg.focus();
  };

  useEffect(() => {}, [socket]);

  const classes = useStyles();
  return (
    <>
      <div className="messenger">
        <div className="chatMenu">
          <div className="chatMenuWrapper">
            <h4 className="chatMenuInput"> Barefoot Nomad Live Chat </h4>
            <Conversation />
          </div>
        </div>
        <div className="chatBox">
          <div className="chatBoxWrapper">
            <div className="chatBoxTop">
              <Message />
              <Message own />
              <Message />
              <Message own />
              <Message />
              <Message own />
              <Message />
              <Message own />
              <Message />
            </div>
            <div className="chatBoxBottom">
              <Formik onSubmit={sendMessage}>
                <Form className={classes.container}>
                  <Grid container spacing={1}>
                    <Grid item xs={10}>
                      <TextField
                        variant="outlined"
                        name="message"
                        placeholder="Type a message..."
                        fullWidth
                        onChange={(e) => setCurrentMessage(e.target.value)}
                      />
                    </Grid>
                    <Grid item xs={2}>
                      <Button
                        variant="outlined"
                        color="primary"
                        onClick={sendMessage}
                        style={{
                          height: '3.4rem',
                          width: '5rem',
                        }}
                      >
                        SEND
                      </Button>
                    </Grid>
                  </Grid>
                </Form>
              </Formik>
            </div>
          </div>
        </div>
        <div className="chatOnline">
          <div className="chatOnlineWrapper">
            <h4 className="chatMenuInput"> Online users </h4>
            <ChatOnline />
          </div>
        </div>
      </div>
    </>
  );
}

export default Chat;
