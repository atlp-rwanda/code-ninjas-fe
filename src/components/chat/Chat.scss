.messenger {
  width: 75vw;
  height: calc(100vh - 80px);
  display: flex;
}
.chatMenu {
  flex: 2.5;
}
.chatMenuWrapper {
  border-right: 1px solid gray;
}
.chatMenuInput {
  width: 90%;
  padding: 5px 0px;
  border-bottom: 1px solid gray;
  margin-top: 2rem;
  color: #1d97a7;
  text-align: center;
}
.chatBox {
  flex: 5.5;
}
.chatBoxWrapper {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  border-right: 1px solid gray;
}
.chatBoxTop {
  height: 100%;
  overflow-y: scroll;
  padding-right: 15px;
}
.chatBoxBottom {
  margin-top: 5px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-top: 30px;
}
.chatMessageInput {
  width: 76%;
  height: 30px;
  padding: 10px;
}
.chatSubmitButton {
  width: 80px;
  height: 40px;
  border: none;
  border-radius: 5px;
  background-color: #1565c0;
  color: white;
  cursor: pointer;
}
.chatOnline {
  flex: 2;
}
.chatMenuWrapper,
.chatBoxWrapper,
.chatOnlineWrapper {
  padding: 10px;
  background: #fafafa;
  height: 100%;
}
