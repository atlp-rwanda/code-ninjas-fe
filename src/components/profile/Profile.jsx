/* eslint-disable no-unused-expressions */
import React, { useState, useEffect } from 'react';
import { Formik, Form } from 'formik';
import * as yup from 'yup';
import { makeStyles } from '@material-ui/core/styles';
import {
  Grid,
  Typography,
  CardActions,
  Button,
  MenuItem,
  TextField,
  Select,
  InputLabel,
} from '@material-ui/core';
import { useDispatch, useSelector } from 'react-redux';
import {
  getProfileData,
  thisUser,
} from '../../redux/features/profile/profileSlice';
import { updateProfileData } from '../../redux/features/profile/updateProfile';
import api from '../../utility/api';

const useStyles = makeStyles((theme) => ({
  formWrapper: {
    marginTop: theme.spacing(5),
    marginBottom: theme.spacing(8),
  },
  actions: {
    justifyContent: 'center',
    padding: '2rem',
    marginTop: theme.spacing(2),
  },
  save: {
    width: '8rem',
    marginRight: theme.spacing(12),
    border: '1px solid #3f51b5',
  },
  cancel: {
    width: '8rem',
    marginRight: theme.spacing(12),
    border: '1px solid #f50057',
  },
  container: {
    backgroundColor: '#fafafa',
    padding: '4rem',
    position: 'relative',
    top: '5%',
    left: '17%',
    width: '76%',
  },
}));

function FormikForm() {
  const dispatch = useDispatch();
  const [isCompleted, setIsComplete] = useState(false);

  useEffect(() => {
    const renderState = async () => {
      api.get('http://localhost:3000/api/users/profile').then((res) => {
        const profile = res.data.body;

        dispatch(getProfileData(profile));
        setIsComplete(res.data.body.isComplete);
        console.log(
          'This is response for profile status ---->>>>>>',
          res.data.body.isComplete
        );
      });
    };
    renderState();
  }, []);

  if (isCompleted) {
    document.getElementById('save').innerText = 'UPDATE';
  }

  const { profile } = useSelector(thisUser);
  console.log('This is redux data ---->>>>>>', profile);

  const handleChange = (e) => {
    console.log('This is new redux data ---->>>>>>', profile);
    dispatch(
      getProfileData({
        ...profile.payload,
        [e.target.name]: e.target.value,
      })
    );
  };

  const handleReset = (e) => {
    e.preventDefault();
    window.location.reload();
  };

  const handleComplete = async () => {
    const {
      gender,
      nationality,
      department,
      preferredLanguage,
      preferredCurrency,
      lineManager,
      address,
      phoneNumber,
      maritalStatus,
      dob,
    } = profile.payload;
    try {
      const response = api.put(
        'http://localhost:3000/api/users/profile/complete',
        {
          gender,
          nationality,
          department,
          preferredLanguage,
          preferredCurrency,
          lineManager,
          address,
          phoneNumber,
          maritalStatus,
          dob,
        }
      );
      console.log('This is response for complete ---->>>>>>', response.data);
      dispatch(updateProfileData(response.data));
      window.alert('Profile Completed Successfully!');
    } catch (err) {
      if (!err?.response) {
        console.log('No Server Response', err);
      } else if (err.response?.status === 422) {
        console.log('Validation Error');
      } else {
        console.log('Profile update Failed');
      }
    }
  };

  const handleUpdate = async () => {
    const {
      gender,
      nationality,
      department,
      preferredLanguage,
      preferredCurrency,
      lineManager,
      address,
      phoneNumber,
      maritalStatus,
      dob,
    } = profile.payload;
    try {
      const response = api.patch(
        'http://localhost:3000/api/users/profile/update',
        {
          gender,
          nationality,
          department,
          preferredLanguage,
          preferredCurrency,
          lineManager,
          address,
          phoneNumber,
          maritalStatus,
          dob,
        }
      );
      console.log('This is response for update ---->>>>>>', response.data);

      dispatch(updateProfileData(response.data));
      window.alert('Profile Updated Successfully');
    } catch (err) {
      const { error } = err.response.data;
      const displayError = document.getElementsByClassName('displayer');
      displayError[0].innerText = error;
      if (!err?.response) {
        console.log('No Server Response', err.response);
      } else if (err.response?.status === 422) {
        console.log('Validation Error');
      } else {
        console.log('Profile update Failed');
      }
    }
  };

  const FORM_COMPLETE_VALIDATION = yup.object().shape({
    gender: yup
      .mixed()
      .oneOf(['male', 'female', 'prefer not say'])
      .required('Please Fill all the fields!'),
    nationality: yup.string().required('Please Fill all the fields!'),
    department: yup.string().required('Please Fill all the fields!'),
    address: yup.string().required('Please Fill all the fields!'),
    lineManager: yup.string().required('Please Fill all the fields!'),
    preferredLanguage: yup.string().required('Please Fill all the fields!'),
    preferredCurrency: yup.string().required('Please Fill all the fields!'),
    phoneNumber: yup
      .number()
      .integer()
      .typeError('Please enter a valid number')
      .required('Please Fill all the fields!'),
    maritalStatus: yup
      .mixed()
      .oneOf(['Single', 'Married', 'Widowed'])
      .required('Please Fill all the fields!'),
    dob: yup
      .date()
      .min('12-31-1800')
      .max('12-31-2010')
      .required('Please Fill all the fields!'),
  });

  const FORM_VALIDATION = yup.object().shape({
    gender: yup.mixed().oneOf(['male', 'female', 'prefer not say']),
    nationality: yup.string(),
    department: yup.string(),
    address: yup.string(),
    lineManager: yup.string(),
    preferredLanguage: yup.string(),
    preferredCurrency: yup.string(),
    phoneNumber: yup
      .number()
      .integer()
      .typeError('Please enter a valid number'),
    maritalStatus: yup.mixed().oneOf(['Single', 'Married', 'Widowed']),
    dob: yup.date().min('12-31-1800').max('12-31-2010'),
  });

  const classes = useStyles();
  return (
    <>
      <Formik
        initialValues={{ ...profile }}
        validationSchema={
          isCompleted ? FORM_COMPLETE_VALIDATION : FORM_VALIDATION
        }
        onSubmit={!isCompleted ? handleComplete : handleUpdate}
      >
        <Form className={classes.container}>
          <Typography className="displayer" />
          <Grid
            container
            spacing={2}
            sx={{
              color: 'red',
              '& .MuiGrid-container': { with: '300rem', color: 'red' },
            }}
          >
            <Grid item xs={12}>
              <Typography>Account Details</Typography>
            </Grid>
            <Grid item xs={6}>
              <TextField
                variant="outlined"
                name="firstName"
                value={profile.payload.firstName}
                label="First Name"
              />
            </Grid>
            <Grid item xs={6}>
              <TextField
                variant="outlined"
                name="lastName"
                value={profile.payload.lastName}
                label="Last Name"
              />
            </Grid>
            <Grid item xs={6}>
              <TextField
                variant="outlined"
                name="email"
                value={profile.payload.email}
                label="Email"
              />
            </Grid>
            <Grid item xs={6}>
              <InputLabel id="Gender"> Gender </InputLabel>
              <Select
                defaultValue={`${profile.payload.gender}`}
                name="gender"
                onChange={handleChange}
                variant="outlined"
                labelId="Gender"
              >
                <MenuItem value="male">Male</MenuItem>
                <MenuItem value="female">Female</MenuItem>
                <MenuItem value="prefer not say">Prefer not say</MenuItem>
              </Select>
            </Grid>
            <Grid item xs={6}>
              <TextField
                variant="outlined"
                name="nationality"
                label="Nationality"
                defaultValue={`${profile.payload.nationality}`}
                onChange={handleChange}
              />
            </Grid>
            <Grid item xs={6}>
              <TextField
                variant="outlined"
                name="department"
                label="Department"
                defaultValue={`${profile.payload.department}`}
                onChange={handleChange}
              />
            </Grid>
            <Grid item xs={6}>
              <TextField
                variant="outlined"
                name="preferredLanguage"
                label="Preferred Language"
                defaultValue={`${profile.payload.preferredLanguage}`}
                onChange={handleChange}
              />
            </Grid>
            <Grid item xs={6}>
              <TextField
                variant="outlined"
                name="preferredCurrency"
                label="Preferred Currency"
                defaultValue={`${profile.payload.preferredCurrency}`}
                onChange={handleChange}
              />
            </Grid>
            <Grid item xs={6}>
              <TextField
                variant="outlined"
                name="lineManager"
                label="Line Manager"
                defaultValue={`${profile.payload.lineManager}`}
                onChange={handleChange}
              />
            </Grid>
            <Grid item xs={6}>
              <TextField
                variant="outlined"
                name="address"
                label="Address"
                defaultValue={`${profile.payload.address}`}
                onChange={handleChange}
              />
            </Grid>
            <Grid item xs={6}>
              <InputLabel id="Marital-Status"> Marital Status</InputLabel>
              <Select
                defaultValue={`${profile.payload.maritalStatus}`}
                name="maritalStatus"
                onChange={handleChange}
                variant="outlined"
                id="Marital-Status"
                labelId="Marital-Status"
              >
                <MenuItem value="Single">Single</MenuItem>
                <MenuItem value="Married">Married</MenuItem>
                <MenuItem value="Widowed">Widowed</MenuItem>
              </Select>
            </Grid>
            <Grid item xs={6}>
              <TextField
                variant="outlined"
                name="phoneNumber"
                label="Phone Number"
                defaultValue={`${profile.payload.phoneNumber}`}
                onChange={handleChange}
              />
            </Grid>
            <Grid item xs={6}>
              <TextField
                id="date"
                label="Date of Birth"
                type="date"
                name="dob"
                defaultValue={`${profile.payload.dob}`}
                format="yyyy/MM/dd"
                min="1900-12-31"
                max="2010-12-31"
                onChange={handleChange}
                InputLabelProps={{
                  shrink: true,
                }}
              />
            </Grid>
            <Grid item xs={12} center>
              <CardActions className={classes.actions}>
                {/* <SubmitButton
                  type="submit"
                  color="primary"
                  className={classes.save}
                >
                  SAVE
                </SubmitButton> */}

                <Button
                  id="save"
                  typeof="submit"
                  color="primary"
                  variant="contained"
                  className={classes.save}
                  onClick={isCompleted ? handleUpdate : handleComplete}
                >
                  COMPLETE
                </Button>

                <Button
                  typeof="button"
                  color="secondary"
                  className={classes.cancel}
                  onClick={handleReset}
                >
                  CANCEL
                </Button>
              </CardActions>
            </Grid>
          </Grid>
        </Form>
      </Formik>
    </>
  );
}

export default FormikForm;
