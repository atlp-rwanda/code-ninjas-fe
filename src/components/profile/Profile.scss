@import url('https://fonts.google.com/css2?family=Montserrat:wght@100:400;800&display=swap');
$bgColor: #1565c0;
// * {
//   box-sizing: border-box;
//   margin: 0;
//   padding: 0;
//   font-family: 'Montserrat', sans-serif;
// }

// body {
//   background-color: #f5f5f5;
//   margin: 0;
//   padding: 0;
// }
// h1 {
//   position: relative;
//   left: -12%;
//   margin-top: 2rem;
//   text-align: center;
//   color: #000;
//   font-size: 16px;
// }
// .header {
//   top: 0;
// }
// label {
//   color: $bgColor !important;
//   margin: 1rem 0 0 0;
// }
// .container {
//   display: flex;
//   justify-content: center;
//   align-items: center;
//   flex-direction: column;
//   margin-left: 10rem;
// }
// // input {
// //   width: 245px;
// //   border: none;
// //   outline: none;
// //   border-bottom: 1px solid #000;
// //   background: transparent;
// //   margin-bottom: 1rem;
// //   padding: 1rem 0 7px 5px;
// //   color: rgb(78, 78, 78);
// // }
// // .info {
// //   display: grid;
// //   grid-template-columns: repeat(2, 1fr);
// //   grid-auto-rows: minmax(12px, auto);
// //   align-items: flex-start;
// //   justify-content: flex-start;
// //   margin-top: 5rem;
// // }
// // .info > div {
// //   display: flex;
// //   flex-direction: column;
// //   // align-items: flex-start;
// //   justify-content: flex-start;
// //   margin: 0 2rem;
// // }
// .image-container {
//   position: relative;
//   left: -10%;
// }
// .profile-image {
//   width: 200px;
//   height: 200px;
//   border-radius: 50%;
//   margin-top: 2rem;
//   margin-bottom: 2rem;
//   background-color: rgb(197, 197, 197);
// }
// span {
//   position: relative;
//   left: -12%;
//   color: #000;
//   margin-top: 3rem;
// }
// .userdata {
//   position: initial !important;
//   color: #1565c0;
//   margin-left: 2px !important;
// }
// select {
//   margin-top: 1rem;
//   outline: none;
//   width: 120px;
//   height: 20px;
//   background: rgb(242, 240, 240);
//   border-radius: 3px;
// }
// ::placeholder {
//   color: #000;
// }
// .image-btn {
//   height: 28px;
//   padding: 5px;
//   position: relative;
//   left: -10%;
//   cursor: pointer;
// }
.button {
  display: flex !important;
  margin: 4rem 0 0 0;
  flex-direction: row !important;
  position: relative;
  left: 25%;
}
.update,
.cancel {
  width: 220px;
  height: 50px;
  border: none;
  outline: none;
  border-radius: 5px;
  background-color: #000;
  color: #fff;
  cursor: pointer;
  font-weight: bold;
  font-size: 18px;
  margin: 1rem;
  letter-spacing: 1.2px;
  padding: 0;
  left: 15%;
}
.update {
  background-color: $bgColor;
  color: #fff;
}
.cancel {
  border: 0.3px solid #000;
  background-color: transparent;
  color: $bgColor;
}
