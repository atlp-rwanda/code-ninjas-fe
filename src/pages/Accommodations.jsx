import React from 'react';
// material
import { Container, Typography } from '@mui/material';
// components
import Page from '../components/dashboard/Page';
import Accomodation from '../components/accommodation/accomodation';

export default function Accommodations() {
  return (
    <Page title="Dashboard: Products">
      <Container>
        <Typography variant="h4" sx={{ mb: 5 }}>
          Accommodations
        </Typography>
        <Accomodation />
      </Container>
    </Page>
  );
}
