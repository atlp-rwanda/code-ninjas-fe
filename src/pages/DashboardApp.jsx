import React from 'react';
import { Container, Typography } from '@mui/material';
// components
import Page from '../components/dashboard/Page';

export default function DashboardApp() {
  return (
    <Page title="Dashboard">
      <Container maxWidth="xl">
        <Typography variant="h4" sx={{ mb: 5 }}>
          Hi, Welcome back
        </Typography>
        <Typography variant="body1" sx={{ mb: 5 }}>
          Dashboard View
        </Typography>
      </Container>
    </Page>
  );
}
