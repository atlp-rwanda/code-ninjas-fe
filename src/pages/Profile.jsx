import React from 'react';
// material
import { Container, Stack, Typography } from '@mui/material';
// components
import Page from '../components/dashboard/Page';

export default function Blog() {
  return (
    <Page title="Dashboard: Blog">
      <Container>
        <Stack
          direction="row"
          alignItems="center"
          justifyContent="space-between"
          mb={5}
        >
          <Typography variant="h4" gutterBottom>
            Profile
          </Typography>
        </Stack>
      </Container>
    </Page>
  );
}
