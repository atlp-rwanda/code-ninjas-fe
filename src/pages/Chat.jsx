import React from 'react';
// material
import { Container, Stack } from '@mui/material';
// components
import Page from '../components/dashboard/Page';
import Chat from '../components/chat/Chat';

export default function Blog() {
  return (
    <Page title="Dashboard: Blog">
      <Container>
        <Stack
          direction="row"
          alignItems="center"
          justifyContent="space-between"
          mb={5}
        >
          <Chat />
        </Stack>
      </Container>
    </Page>
  );
}
