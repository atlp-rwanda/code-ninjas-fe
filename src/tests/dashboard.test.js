import React from 'react';
import { screen, render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import DashboardApp from '../pages/DashboardApp';
import Chat from '../pages/Chat';
import Accommodations from '../pages/Accommodations';
import Profile from '../pages/Profile';
import Users from '../pages/Users';
import TripRequest from '../pages/TripRequest';

describe('Dashboard Page Component', () => {
  beforeAll(() => {
    render(<DashboardApp />);
  });

  it('should have the right message in the dom', () => {
    const message = 'Hi, Welcome back';

    expect(screen.getByText(message)).toBeInTheDocument();
  });

  it('Should Take Snapshot for Home Component', () => {
    const renderedComponent = renderer.create(<DashboardApp />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  afterAll(cleanup);
});

describe('Dashboard Accommodations Page Component', () => {
  beforeAll(() => {
    render(<Accommodations />);
  });

  it('should have the right message in the dom', () => {
    const message = 'Accommodations';

    expect(screen.getByText(message)).toBeInTheDocument();
  });

  it('Should Take Snapshot for Home Component', () => {
    const renderedComponent = renderer.create(<Accommodations />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  afterAll(cleanup);
});

describe('Dashboard Chat Page Component', () => {
  beforeAll(() => {
    render(<Chat />);
  });

  it('should have the right message in the dom', () => {
    const message = 'Chat';

    expect(screen.getByText(message)).toBeInTheDocument();
  });

  it('Should Take Snapshot for Home Component', () => {
    const renderedComponent = renderer.create(<Chat />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  afterAll(cleanup);
});

describe('Dashboard Profile Page Component', () => {
  beforeAll(() => {
    render(<Profile />);
  });

  it('should have the right message in the dom', () => {
    const message = 'Profile';

    expect(screen.getByText(message)).toBeInTheDocument();
  });

  it('Should Take Snapshot for Home Component', () => {
    const renderedComponent = renderer.create(<Profile />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  afterAll(cleanup);
});
