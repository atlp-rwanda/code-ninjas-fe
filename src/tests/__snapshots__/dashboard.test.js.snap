// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Accommodations Page Component Should Take Snapshot for Home Component 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 css-et7x9c-MuiTypography-root"
    >
      Accommodations
    </h4>
    <div
      className="css-noiw7-MuiStack-root"
      data-testid="AccommodationscardContainer"
    />
  </div>
</div>
`;

exports[`Dashboard Chat Page Component Should Take Snapshot for Home Component 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="css-d3ri6l-MuiStack-root"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
      >
        Chat
      </h4>
    </div>
  </div>
</div>
`;

exports[`Dashboard Page Component Should Take Snapshot for Home Component 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 css-et7x9c-MuiTypography-root"
    >
      Hi, Welcome back
    </h4>
    <p
      className="MuiTypography-root MuiTypography-body1 css-17665zs-MuiTypography-root"
    >
      Dashboard View
    </p>
  </div>
</div>
`;
